src/main/java/org/jboss/tools/intellij/analytics/PreloadLanguageServer.java

```diff
@@ -14,10 +16,13 @@
   private final ICookie cookies = ServiceManager.getService(Settings.class);
 
   private void attachLanguageClient(final File cliFile) {
-    final String[] EXTENSIONS = {"xml", "json", "txt"};
+    List<String> supportedExtensions = Platform.supportedManifestFiles;
     final String[] cmds = {cliFile.toString(), "--stdio"};
     ApplicationManager.getApplication().invokeAndWait(() -> {
-      for (String ext : EXTENSIONS) {
+      // Run loop for -1 times because requirements and requirements-dev are having same extension
```

Comment:
That is not good if you update the list then this may fail. Use Streams.distincts to remove duplicates

Reply:
Yes I agree, I was thinking that as of now we have no plan to add new ecosystem, so this simple approach should be fine. I will change it.